app/pages/api/realms/[id].ts

```diff
@@ -0,0 +1,185 @@
+        updateRequest.division =
+          updateRequest.division === 'Other' ? updateRequest.divisionOther : updateRequest.division;
+        updateRequest.branch = updateRequest.branch === 'Other' ? updateRequest.branchOther : updateRequest.branch;
+
+        currentRequest = await prisma.roster.findUnique({
+          where: {
+            id: parseInt(req.query.id as string, 10),
+          },
+        });
+
+        if (!currentRequest) {
+          return res.status(400).json({ success: false, error: 'Invalid request' });
+        } else if (currentRequest.status !== StatusEnum.PENDING && updateRequest.approved) {
+          return res.status(400).json({ success: false, error: 'Request is in invalid stage' });
+        }
+
+        isPO = username.toLowerCase() === currentRequest.productOwnerIdirUserId?.toLowerCase();
+
+        updaterRole = isAdmin ? RoleEnum.ADMIN : isPO ? RoleEnum.PRODUCT_OWNER : RoleEnum.TECHNICAL_LEAD;
+
+        try {
+          updateRequest = getUpdateRealmSchemaByRole(updaterRole).validateSync(updateRequest, {
+            abortEarly: false,
+            stripUnknown: true,
+          });
+        } catch (e) {
+          const error = e as ValidationError;
+          return res.status(400).json({ success: false, error: error.errors });
+        }
+
+        if (isAdmin) {
+          if (
+            currentRequest.status === StatusEnum.PENDING &&
+            String(updateRequest.approved) === 'true' &&
+            !currentRequest.prNumber &&
+            !currentRequest.approved
+          ) {
+            await createEvent({
+              realmId: parseInt(req.query.id as string, 10),
+              eventCode: EventEnum.REQUEST_APPROVE_SUCCESS,
+              idirUserId: username,
+              details: req.body,
+            });
+            const prResponse: CreatePullRequestResponseType = await createCustomRealmPullRequest(
+              currentRequest.realm!,
+              currentRequest.environments,
+            );
+            const pr = await mergePullRequest(prResponse.data.number);
+            if (pr.data.merged) await deleteBranch(currentRequest.realm!);
+            updateRequest.prNumber = prResponse.data.number;
+            updateRequest.status = updateRequest.prNumber ? StatusEnum.PRSUCCESS : StatusEnum.PRFAILED;
+            updateRequest.approved = true;
+          } else if (
+            currentRequest.status === StatusEnum.PENDING &&
+            String(updateRequest.approved) === 'false' &&
+            !currentRequest.prNumber
+          ) {
+            await createEvent({
+              realmId: parseInt(req.query.id as string, 10),
+              eventCode: EventEnum.REQUEST_REJECT_SUCCESS,
+              idirUserId: username,
+              details: req.body,
+            });
+            updateRequest.approved = false;
+          }
+        }
+
+        updatedRealm = await prisma.roster.update({
```

Comment:
Does the put request validate ownership? Not sure if its in that `getUpdateRealmSchemaByRole` funtion

Reply:
Only logged in users would only be able to view their owned realms but I have updated the where condition to be more descriptive.